src/coord/src/coord.rs

```diff
@@ -332,6 +334,49 @@ fn concretize_replica_config(
     Ok(config)
 }
 
+/// Holds tables needing advancement.
+struct AdvanceTables<T> {
+    map: HashMap<GlobalId, T>,
```

Comment:
Is there ever a scenario where the values would not all be the same timestamp? If not it might make sense to change this to have a `Set<GlobalId>` and single `T` field.